### PR TITLE
[IMP] luxon update + account for local timezone when formatting datetimes + mockserver fix

### PR DESCRIPTION
**Pull request by brboi**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by brboi** [comment]
luxon upgrade only is green: https://runbot.odoo.com/runbot/batch/821417

File: addons/web/static/src/views/relational_model.js [hunk 1]
```diff
@@ -661,10 +661,12 @@ export class Record extends DataPoint {
             } else if (fieldType === "many2one") {
                 changes[fieldName] = changes[fieldName] ? changes[fieldName][0] : false;
             } else if (fieldType === "date") {
-                changes[fieldName] = changes[fieldName] ? serializeDate(changes[fieldName]) : false;
+                changes[fieldName] = changes[fieldName]
+                    ? serializeDate(changes[fieldName].setZone("utc", { keepLocalTime: true }))
```
Comment (brboi) [inline addons/web/static/src/views/relational_model.js:665]: useless ?

File: addons/web/static/src/views/basic_relational_model.js [hunk 2]
```diff
@@ -30,10 +30,16 @@ function mapWowlValueToLegacy(value, type) {
     switch (type) {
         case "date":
             // from luxon to moment
-            return value ? parseDate(serializeDate(value), null, { isUTC: true }) : false;
+            return value
+                ? parseDate(serializeDate(value.setZone("utc", { keepLocalTime: true })), null, {
+                      isUTC: true,
+                  })
+                : false;
```
Comment (brboi) [inline addons/web/static/src/views/basic_relational_model.js:37]: useless ?

File: addons/web/static/src/search/filter_menu/custom_filter_item.js [hunk 3]
```diff
@@ -221,7 +221,7 @@ export class CustomFilterItem extends Component {
                 // No description to push here
             } else if (["date", "datetime"].includes(genericType)) {
                 const serialize = genericType === "date" ? serializeDate : serializeDateTime;
-                domainValue = condition.value.map(serialize);
+                domainValue = condition.value.map((v) => serialize(v.setZone("utc")));
```
Comment (brboi) [inline addons/web/static/src/search/filter_menu/custom_filter_item.js:224]: maybe setZone("utc") in serialize directly

**Review by brboi — COMMENTED** [review]
make the second commit an IMP + explain

**Review by aab-odoo — APPROVED** [review]

**Comment by brboi** [comment]
robodoo rebase-ff

**Comment by aab-odoo** [comment]
robodoo rebase-ff

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**